components/Shell.tsx

```diff
@@ -12,16 +13,26 @@ export default function Shell(props) {
   const [mobileMenuExpanded, setMobileMenuExpanded] = useState(false);
   const telemetry = useTelemetry();
 
+  const handleResizeEvent = () => setProfileDropdownExpanded(false);
+
+  useEffect(() => {
+    if (profileDropdownExpanded) {
+      window.addEventListener("resize", handleResizeEvent);
+    } else {
+      window.removeEventListener("resize", handleResizeEvent);
+    }
+
+    return () => {
+      window.removeEventListener("resize", handleResizeEvent);
+    };
+  }, [profileDropdownExpanded]);
```

Comment:
This effect listens for the `resize` event only when the menu is opened - then if the window resizes it immediately closes it and removes the listener.